src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs

```diff
@@ -645,6 +645,9 @@ public virtual void Emit (OpCode opcode, LocalBuilder local)
 				throw new ArgumentException ("Trying to emit a local from a different ILGenerator.");
 
 			uint pos = local.position;
+			if ((opcode == OpCodes.Ldloca_S || opcode == OpCodes.Ldloc_S || opcode == OpCodes.Stloc_S) && pos >= 255)
+				throw new InvalidOperationException ("Opcodes using a short-form index cannot address a local position over 255.");
```

Comment:
is 255 a valid position? you're checking for `pos >= 255` above

Reply:
Fixed, thanks!